src/kvstore/kvstore_dist_server.h

```diff
@@ -428,20 +468,42 @@ class KVStoreDistServer {
   }
 
   /**
-   * \brief user defined
+   * \brief user defined mode for push
    */
   bool sync_mode_;
   KVStore::Controller controller_;
   KVStore::Updater updater_;
 
+  /**
+   * \brief store_ contains the value at kvstore for each key
+   */
   std::unordered_map<int, NDArray> store_;
+
+  /**
+   * \brief merge_buf_ is a buffer used if sync_mode is true. It represents
+   * values from different workers being merged. The store will be updated
+   * to this value when values from all workers are pushed into this buffer.
+   */
   std::unordered_map<int, MergeBuf> merge_buf_;
 
+  /**
+   * \brief decomp_buf_ is a buffer into which compressed values are
+   * decompressed before merging to the store. used when compress_!='none'
+   */
+  std::unordered_map<int, NDArray> decomp_buf_;
+
   Executor exec_;
   ps::KVServer<float>* ps_server_;
 
   // whether to LOG verbose information
   bool log_verbose_;
+
+  /**
+   * \brief compress_ refers to whether values sent to kvstore server are
+   * in quantized form. It can be 'none' or '2bit' for now. This is set
+   * by worker by sending a command to server
+   */
+  std::string compress_ = "none";
```

Comment:
How often is this accessed? Would it be more efficient to convert the initial string to an enum?